airbyte-integrations/connectors/source-trustpilot/source_trustpilot/trustpilot.yaml

```
@@ -0,0 +1,60 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://www.trustpilot.com"
+    http_method: "GET"
+    authenticator:
+      type: NoAuth
+    request_options_provider:
+      request_parameters:
+        languages: all
+        sort: recency
+  retriever:
+    record_selector:
+      extractor:
+        class_name: "source_trustpilot.source.TrustpilotExtractor"
+      record_filter:
+        condition: "{{ record['datePublished'] > stream_state['old_datePublished'] }}"
+    paginator:
+      class_name: "source_trustpilot.source.TrustpilotPaginator"
+      url_base: "*ref(definitions.requester.url_base)"
+      # `page_size_option` is dummy to support DefaultPaginator API
+      page_size_option:
+        inject_into: "request_parameter"
+        field_name: "page_size"
```

```suggestion
        field_name: "page"
```
Did you try to change this to not have a custom code?

---

The problem is not directed to `page_size` but to `page` which starts from 0 and then goes to 1, 2, etc.
Truspilot paginations starts with 1 (empty page) and then goes to 2, 3, etc.